### PR TITLE
Added Psalm and improved CI

### PR DESCRIPTION
**Pull request by Koen1999**

### CODE REVIEW

File: .github/workflows/static-analysis-psalm.yml [hunk 1]
```diff
@@ -0,0 +1,102 @@
+name: PHP Static Analysis with Psalm
```
Comment (tomudding) [inline .github/workflows/static-analysis-psalm.yml:1]: We might want to split these two static analysis actions into another action (or maybe two) that acts as a composite action where we do the setup.

**Comment by Koen1999** [comment]
@tomudding 

I set an appropriate error level for the project and generated a baseline containing some issues.
The Github action will show new errors, warnings, and informational notices introduced in the PR similar to what PHPStan does.

From what I have seen Psalm has some valid points and actually more than PhpStan at our current error level settings.

This PR also solved some small issues.

Moreover, static code analysis now also covers the test source files.

File: .github/workflows/static-analysis-psalm.yml (hunk 1 above)
Comment (Koen1999) [reply .github/workflows/static-analysis-psalm.yml:1]: Performance wise this is probably a good idea. But I'm not that good with github actions

**Comment by Koen1999** [comment]
You can see an example PR here: https://github.com/Koen1999/gewisweb/pull/8

File: .github/workflows/static-analysis-psalm.yml (hunk 1 above)
Comment (tomudding) [reply .github/workflows/static-analysis-psalm.yml:1]: We can leave it like this for now, we just have to make sure that if we change something in the duplicate code we must change it in both actions.

**Comment by Koen1999** [comment]
It works as expected now :D

File: module/Activity/src/Controller/ActivityController.php [hunk 2]
```diff
@@ -290,6 +291,7 @@ public function signupAction(): Response|ViewModel
 
         if ($request->isPost()) {
             $form = $this->signupService->getForm($signupList);
+            /** @var ParametersInterface $postData */
             $postData = $request->getPost();
```
Comment (tomudding) [inline module/Activity/src/Controller/ActivityController.php:295]: This structure is used in many places (not everywhere) so we might want to do this in more places in the future.

Comment (Koen1999) [reply module/Activity/src/Controller/ActivityController.php:295]: Yes, but somehow these specific cases were problematic according to the static analysis

**Pull request merged**